pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/BaseChunkSingleValueReader.java

```diff
@@ -66,18 +70,49 @@ public BaseChunkSingleValueReader(PinotDataBuffer pinotDataBuffer, ChunkDecompre
 
     _lengthOfLongestEntry = _dataBuffer.getInt(headerOffset);
     headerOffset += INT_SIZE;
+
+    int dataHeaderStart = headerOffset;
+    if (version > 1) {
```

Comment:
if this is not widely used. Get rid of the initial version and write a converter from v1 to v2. I prefer keeping the runtime code simpler and fix everything as part of loading the segment if possible.

Reply:
Agree with that. Unfortunately, It is used in production already, and would require a few TB of data to be converted.